### PR TITLE
Entry flow UX changes - Part 2

### PR DESCRIPTION
**Pull request by brianpeiris**
See [commit list](https://github.com/mozilla/mr-social-client/pull/177/commits) for changes. 
For #116. Building on #175.

### CODE REVIEW

File: src/react-components/profile-entry-panel.js [hunk 1]
```diff
@@ -75,20 +78,47 @@ class ProfileEntryPanel extends Component {
...
+                title={formatMessage({ id: "profile.display_name.validation_warning" })}
+                ref={inp => (this.nameInput = inp)}
+              />
+            </label>
             <iframe
               className="profile-entry__avatar-selector"
               src={`/${this.props.htmlPrefix}avatar-selector.html#avatar_id=${this.state.avatar_id}`}
               ref={ifr => (this.avatarSelector = ifr)}
             />
+            {!this.props.store.state.profile.has_agreed_to_terms && (
```
Comment (brianpeiris) [inline src/react-components/profile-entry-panel.js]: @gfodor does it make sense to hide the term checkbox after the user has agreed to it?

File: src/assets/translations.data.json [hunk 2]
```diff
@@ -14,8 +14,14 @@
...
     "entry.enable-screen-sharing": "Share my desktop",
     "profile.save": "SAVE",
+    "profile.display_name.label": "Display name:",
     "profile.display_name.validation_warning": "Alphanumerics and hyphens. At least 3 characters, no more than 32",
     "profile.header": "Your identity",
+    "profile.terms.prefix": "I confirm that I am over the age of 13 and agree to the",
+    "profile.terms.privacy": "privacy policy",
+    "profile.terms.conjunction": "and",
+    "profile.terms.tou": "terms of use",
+    "profile.terms.suffix": ".",
```
Comment (brianpeiris) [inline src/assets/translations.data.json]: @gfodor Terms copy needs your review.

File: src/react-components/profile-entry-panel.js [hunk 3]
```diff
@@ -75,20 +78,47 @@ class ProfileEntryPanel extends Component {
...
+                />
+                <span className="profile-entry__terms__text">
+                  <FormattedMessage id="profile.terms.prefix" />{" "}
+                  <a className="profile-entry__terms__link" target="_blank" href="/privacy">
+                    <FormattedMessage id="profile.terms.privacy" />
+                  </a>{" "}
+                  <FormattedMessage id="profile.terms.conjunction" />{" "}
+                  <a className="profile-entry__terms__link" target="_blank" href="/terms">
+                    <FormattedMessage id="profile.terms.tou" />
+                  </a>
```
Comment (brianpeiris) [inline src/react-components/profile-entry-panel.js]: @gfodor Linking to non-existent `/privacy` and `/terms` urls for now.

File: src/assets/translations.data.json (hunk 2 above)
Comment (gfodor) [reply src/assets/translations.data.json]: yes we can send this to legal once they get things wrapped up

File: src/react-components/profile-entry-panel.js (hunk 1 above)
Comment (gfodor) [reply src/react-components/profile-entry-panel.js]: i think so

File: src/hub.js [hunk 4]
```diff
@@ -243,26 +240,33 @@ const onReady = async () => {
...
+  mountUI(scene);
+
+  const remountUI = () => {
+    mountUI(scene);
+  };
 
   getAvailableVREntryTypes().then(availableVREntryTypes => {
-    uiRoot.setState({ availableVREntryTypes });
-    uiRoot.handleForcedVREntryType();
+    console.log('BPDEBUG availableVREntryTypes', availableVREntryTypes);
```
Comment (gfodor) [inline src/hub.js]: log

File: src/hub.js [hunk 5]
```diff
@@ -243,26 +240,33 @@ const onReady = async () => {
 
   registerNetworkSchemas();
 
-  const uiRoot = mountUI(scene);
+  mountUI(scene);
+
+  const remountUI = () => {
```
Comment (gfodor) [inline src/hub.js]: maybe set things up so instead of having to mutate `uiRootProps` everywhere, the k/vs to alter are arguments to this function?

**Pull request merged**